crates/phactory/src/prpc_service.rs

```diff
@@ -1109,6 +1109,35 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> Phactory<Platform>
         self.cluster_state_to_apply = Some(cluster_state);
         Ok(())
     }
+
+    fn get_worker_key_challenge(
+        &mut self,
+        block_number: chain::BlockNumber,
+        now: u64,
+    ) -> HandoverChallenge<chain::BlockNumber> {
+        let sgx_target_info = if self.dev_mode {
+            vec![]
+        } else {
+            let my_target_info = sgx_api_lite::target_info().unwrap();
+            sgx_api_lite::encode(&my_target_info).to_vec()
+        };
+        let challenge = HandoverChallenge {
+            sgx_target_info,
+            block_number,
+            now,
+            dev_mode: self.dev_mode,
+            nonce: crate::generate_random_info(),
+        };
+        self.handover_last_challenge = Some(challenge.clone());
+        challenge
+    }
+
+    pub fn verify_worker_key_challenge(
+        &mut self,
+        challenge: &HandoverChallenge<chain::BlockNumber>,
+    ) -> bool {
+        return self.handover_last_challenge.take().as_ref() == Some(challenge);
```

Comment:
Now `handover_last_challenge` will not be cleared after the verification.

Reply:
The `take()` would take it away.

Reply:
Got it. I missed it.